### PR TITLE
added simple RabbitMQ client

### LINKED ISSUE #101: v2 - RabbitMq Client

### PR DESCRIPTION
**Pull request by MichalStopyra**

### CODE REVIEW

File: clients/rabbitmq/src/main/scala/mqperf/rabbitmq/RabbitMq.scala [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+  var conn: Connection = _
+
+  override def init(config: Config): Unit = {
+    val host: String = config.mqConfig(HostsConfigKey)
+    val queueName: String = config.mqConfig(QueueNameConfigKey)
+
+    val cf = new ConnectionFactory()
+    cf.setHost(host)
+    cf.setUsername("guest")
```
Comment (micossow) [inline clients/rabbitmq/src/main/scala/mqperf/rabbitmq/RabbitMq.scala]: shouldn't this be configurable?

File: clients/rabbitmq/src/main/scala/mqperf/rabbitmq/RabbitMq.scala [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+import java.util.concurrent.ConcurrentLinkedQueue
+import scala.annotation.tailrec
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.{Future, blocking}
+import scala.util.Try
+
+class RabbitMq extends Mq with StrictLogging {
+  private val HostsConfigKey = "hosts"
+  private val QueueNameConfigKey = "queueName"
+  private val NrPollReattemptConfigKey = "nrPollReattempt"
```
Comment (micossow) [inline clients/rabbitmq/src/main/scala/mqperf/rabbitmq/RabbitMq.scala]: how about `maxPollAttempts`?

File: clients/rabbitmq/src/main/scala/mqperf/rabbitmq/RabbitMq.scala [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    override def send(msgs: Seq[String]): Future[Unit] = Future {
+      msgs.foreach(msg => channel.basicPublish("", queueName, MessageProperties.PERSISTENT_TEXT_PLAIN, msg.getBytes))
+    }
+  }
+
+  override def createReceiver(config: Config): MqReceiver = new MqReceiver {
+    override type MsgId = Long
+
+    private val queueName: String = config.mqConfig(QueueNameConfigKey)
+    private val nrPollReattempt: Int = Try(config.mqConfig(NrPollReattemptConfigKey).toInt).getOrElse(0)
```
Comment (micossow) [inline clients/rabbitmq/src/main/scala/mqperf/rabbitmq/RabbitMq.scala]: should we ignore an error though? I'd use Option instead, for example:
```scala
Option(config.mqConfig(NrPollReattemptConfigKey)).map(_.toInt).getOrElse(0)
```

File: clients/rabbitmq/src/main/scala/mqperf/rabbitmq/RabbitMq.scala [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+          }
+
+          doPoll(if (waitForMsgs) nrPollReattempt else 0)
+        }
+
+        doReceive(Nil, maxMsgCount)
+      }
+    }
+
+    private val multipleAck = Try(config.mqConfig("multipleAck").toBoolean).getOrElse(false)
```
Comment (micossow) [inline clients/rabbitmq/src/main/scala/mqperf/rabbitmq/RabbitMq.scala]: same as above

**Review by micossow — CHANGES_REQUESTED** [review]
Nice job, some minor issues only

File: clients/rabbitmq/src/main/scala/mqperf/rabbitmq/RabbitMq.scala (hunk 2 above)
Comment (MichalStopyra) [reply clients/rabbitmq/src/main/scala/mqperf/rabbitmq/RabbitMq.scala]: sounds, better - I'll switch to that :)

**Review by micossow — APPROVED** [review]

File: clients/rabbitmq/src/main/scala/mqperf/rabbitmq/RabbitMq.scala [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+    channel
+  }
+
+  override def createSender(config: Config): MqSender = new MqSender {
+    private val queueName: String = config.mqConfig(QueueNameConfigKey)
+    private val channel = newChannel(queueName, passive = true)
+    channel.confirmSelect()
+
+    override def send(msgs: Seq[String]): Future[Unit] = Future {
+      msgs.foreach(msg => channel.basicPublish("", queueName, MessageProperties.PERSISTENT_TEXT_PLAIN, msg.getBytes))
```
Comment (adamw) [inline clients/rabbitmq/src/main/scala/mqperf/rabbitmq/RabbitMq.scala]: is this a blocking call?

Comment (adamw) [reply clients/rabbitmq/src/main/scala/mqperf/rabbitmq/RabbitMq.scala]: if available, we should use a non-blocking client

File: clients/rabbitmq/src/main/scala/mqperf/rabbitmq/RabbitMq.scala [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+              case Some(m) => doReceive(acc :+ m, count - 1)
+            }
+          }
+        }
+
+        def nextMessageFromQueue(waitForMsgs: Boolean): Option[(MsgId, String)] = {
+          @tailrec
+          def doPoll(waitIterations: Int): Option[(MsgId, String)] = {
+
+            val next = queue.poll()
```
Comment (adamw) [inline clients/rabbitmq/src/main/scala/mqperf/rabbitmq/RabbitMq.scala]: same here- is there a way to receive in a non-blocking way? preferrably without polling?

**Pull request closed without merging**